### ISSUE TITLE
[1828] Game allows buying 5T from another company after emergency fundraising more than the price of the 5T

### ISSUE THREAD

**Issue by physic** [tobymao/18xx#11333]
~~https://18xx.games/game/179384?action=1017~~

~~Press [Show trains from other players].~~

~~Expected: All trains belonging to other players are shown.~~
~~Observed: 5s, 6s are shown. 3+Ds are not.~~

Relevant Gamestate:
The player is operating the NNH and must buy a train.
The next available train is the 8E.
The NNH treasury and the player have a combined $556.
The player sells a share via emergency fundraising.
The player selects [Show trains from other players].

Steps to Reproduce:
Follow this link, and clone the game. https://18xx.games/game/179384?action=1025
On the games tab, select [Show trains from other players].
You should see something like the below screenshot.

Expected:
- The player must spend $557 or more on a train, to make the share sale legal.
- This rules out a 5 or a 3+D, since they have a face value of $250 and $350 respectively, and when you EMR, you may not pay more than face value for a train.
- The player is able to buy a 6 for between $557 and $650
- The player is able to sell shares to raise additional funds to buy the 8E from the bank for $800.

Observed:
- The player is *not* able to buy a 3+D. ✅
- **The player is able to buy a 5 for $1 or more** ❌
- The player is *not* able to buy a 6 for less than $557. ✅
- The player is able to buy a 6 for between $557 and $650 ✅
- The player is able to sell shares to raise additional funds to buy the 8E from the bank for $800. ✅

![image](https://github.com/user-attachments/assets/93e77377-e9f3-4c2f-ad9f-39b355167c9c)

**Comment by dfannius** [comment]
FWIW, I saw the 3+Ds when I followed the link (I believe you that you didn't):

<img width="294" alt="Screenshot 2024-11-09 at 18 18 04" src="https://github.com/user-attachments/assets/a8ea8d5b-ab0d-4188-9dd3-1c08759f9812">

**Comment by physic** [comment]
Okay. I was confused. You're right. I misunderstood the issue and posted a bad link. 

Let me try again.
https://18xx.games/game/179384?action=1025

The player sold a share emergency fundraising. In order to sell that share, he must spend 557+, which precludes buying a 5 or 3+D, because you may not pay more than the price of the train.

However, the 5's are showing up in the list, and the app allows buying a 5 for $1 after having EMR'ed the share.

**Comment by scottredracecar** [comment]
Yeah, I was always surprised that the site even works as well as it does
for this with the money raising before the train buying.

Intuitively playing tabletop, I always think you pick the train first, then
do the "figure out how to pay for it" part. I think implementing it like
that is a major change in the EMR implementation, but I think it's probably
necessary to cover cases like this.

I don't think the current system is robust enough to handle these
scenarios. Not sure if anyone has the will to change it though.

**Comment by philcampeau** [comment]
It's definitely doable, but will anyone invest the time given that players can follow the rules as written? I's really a question of implementing a bunch of code to prevent cheating.

**Comment by roseundy** [comment]
Does the code really need to change that much if all it does is raise an error when you buy the "wrong" train?

**Comment by physic** [comment]
There seems to be confusion about how much the site is already doing. Follow https://18xx.games/game/179384?action=1025 and then clone the game and you should see the game state I am talking about.

The game engine already knows (1) that you sold a share via EMR when you had $556, so it knows you must spend at least $557. It also seems to know (2) that you cannot buy a 3+D. But for some reason, it still allows buying a 5. Here's a screenshot.

![image](https://github.com/user-attachments/assets/93e77377-e9f3-4c2f-ad9f-39b355167c9c)

**Comment by scottredracecar** [comment]
This is what it looks like when I clone that action (different).

<img width="430" alt="image" src="https://github.com/user-attachments/assets/7dfce98f-e366-4906-829d-0da513241fa2">

**Comment by physic** [comment]
@scottredracecar Sorry I have been sloppy and fragmented about repro steps. If you then [Show trains from other players] you should see the same thing as in my last message.

**Comment by physic** [comment]
I updated the original comment. Hopefully I've captured all of the information in an understandable way.

**Comment by scottredracecar** [comment]
Seems fixable without redoing the interface now that I understand it better.

**Comment by crericha** [comment]
Is this a regression or always like that?

**Comment by physic** [comment]
@crericha I do not know if it is a regression. So far, this case in this game is the only instance of it that I am aware of.

**Comment by physic** [comment]
@philcampeau Thanks! It was a mess of incomplete and not entirely correct information when I wrote it last week, so that's probably on me 😅 Hopefully it makes more sense now.
